### PR TITLE
Issue 434 | Docker compose profiles + versions

### LINKED ISSUE #434: Dockerfile: add version support to dockerfiles in order to use nightly, stable, latest or versioned images
### What is the expected enhancement?

Add version support to dockerfiles in order to use nightly, stable, latest or versioned images

---

By @psschwei 

assuming you meant the compose file, we could do something like:
```yaml
services:
  jupyter:
    container_name: qs-jupyter
    image: qiskit/quantum-serverless-notebook:${VERSION:-0.0.6}-py39
    ...
```

and then if you wanted to run nightly do something like VERSION=nightly docker compose up

### PR DESCRIPTION
**Pull request by IceKhan13**
### Summary

Docker profiles and versions configuration

### Details and comments

For full setup

```shell
VERSION=nightly docker-compose --profile full up
```

For minimal setup (gateway, repo, keycloak, ray)

```shell
VERSION=nightly docker-compose up
```

For minimal setup (gateway, repo, keycloak, ray) + jupyter

```shell
VERSION=nightly docker-compose --profile jupyter up
```

Closes #434

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ You don't have to worry about configuration or scaling up computational resource
 1. Prepare local infrastructure
 ```shell
 docker-compose pull
-docker-compose up
+VERSION=nightly docker-compose --profile jupyter --profile observability up
```
Comment (psschwei) [inline README.md:39]: I think we could add a "full" profile (something like `profiles: ["jupyter", "full"]` in the compose file, which would drop a profile flag in the call: `VERSION=nightly docker-compose --profile full up`

(maybe "all" a the better choice than "full"? don't have a strong preference either way)

**Comment by psschwei** [comment]
I think this also will tackle #372 as well

File: README.md (hunk 1 above)
Comment (IceKhan13) [reply README.md]: I like `full` more :)

**Review by psschwei — APPROVED** [review]
LGTM

**Pull request merged**